osutil/disks/gpt_test.go

```diff
@@ -0,0 +1,247 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package disks_test
+
+import (
+	"encoding/binary"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/osutil/disks"
+)
+
+type gptSuite struct {
+	image string
+	size  uint64
+}
+
+var _ = Suite(&gptSuite{})
+
+func (s *gptSuite) SetUpTest(c *C) {
+	tmpdir := c.MkDir()
+	s.image = filepath.Join(tmpdir, "image.img")
+	err := exec.Command("truncate", "--size", "128M", s.image).Run()
+	c.Assert(err, IsNil)
+	cmd := exec.Command("sfdisk", s.image)
+	cmd.Stdin = strings.NewReader("label: gpt\n")
+	err = cmd.Run()
+	c.Assert(err, IsNil)
```

Comment:
It seems on 16.04 and 14.04, the sfdisk command is not available:
```

----------------------------------------------------------------------
FAIL: gpt_test.go:41: gptSuite.SetUpTest

gpt_test.go:49:
    c.Assert(err, IsNil)
... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc0002cc380), Stderr:[]uint8(nil)} ("exit status 1")


----------------------------------------------------------------------
PANIC: gpt_test.go:178: gptSuite.TestBadCRC

... Panic: Fixture has panicked (see related PANIC)
OOPS: 46 passed, 1 FAILED, 12 MISSED
--- FAIL: Test (0.27s)
FAIL
FAIL	github.com/snapcore/snapd/osutil/disks	0.274s
```

is there anyway we can setup a gpt partition image without using sfdisk this way for testing purposes? it seems we have the following options:

* document that sfdisk is needed to run unit tests and possibly adjust build-deps
* skip this unit test when sfdisk is not available
* embed a generated header / backup header bytes into the test as a []byte variable and write them out to a file for the test

Reply:
I have put the headers as files in testdata and rebuild the image from those. There is a script that can re-generates the files.